### PR TITLE
Remove .envrc requirement

### PR DESCRIPTION
**Pull request by sylv-io**
Move .envrc to setup.env and make it optional.
It is not required to source the environment file to build
all task targets.

Signed-off-by: Marcello Sylvester Bauer <sylv@sylv.io>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -57,18 +57,18 @@ task clean-all
 
 ## Environment
 
-The System Transparency Repository provides a `.envrc` file to load the build environment. It installs the latest version of [Task](https://taskfile.dev) and configures a separate Go environment to prevent any conflicts. To load and unload the environment depending on the current directory, it is recommended to use [direnv](https://direnv.net/). Go to [Basic Installation](https://direnv.net/#basic-installation) to see how to properly setup direnv your shell.
+The System Transparency Repository provides a `setup.env` file to load the build environment. It installs the latest version of [Task](https://taskfile.dev) and configures a separate GOPATH to prevent any conflicts. To load and unload the environment depending on the current directory, it is recommended to use [direnv](https://direnv.net/). Go to [Basic Installation](https://direnv.net/#basic-installation) to see how to properly setup direnv your shell.
```
Comment (jensdrenhaus) [inline README.md:60]: setup direnv FOR you shel

Comment (sylv-io) [reply README.md]: ack

File: setup.env [hunk 2]
```diff
@@ -1,13 +1,12 @@
...
 GOBIN="${GOPATH}/bin"
-TASK="${TASKBIN}/task"
 
 # install task
-[[ -x "${TASK}" ]] || sh -c "$(curl --location https://taskfile.dev/install.sh)" -- -b "${TASKBIN}"
+[[ -x "${TASKBIN}/task" ]] || sh -c "$(curl --location https://taskfile.dev/install.sh)" -- -b "${TASKBIN}"
 
 # export custom go environment
-export GOPATH="${PWD}/cache/go"
+export GOPATH
```
Comment (MDr164) [inline setup.env:9]: Does this actually work? Or does need to look like `export GOPATH="${GOPATH}"`?

Comment (sylv-io) [reply setup.env:9]: Line 2 does already assign a value to GOPATH, and it is optional but not required for exporting:

```
export: export [-fn] [name[=value] ...] or export -p
    Set export attribute for shell variables.
    
    Marks each NAME for automatic export to the environment of subsequently
    executed commands.  If VALUE is supplied, assign VALUE before exporting.
 [...]
```

**Review by MDr164 — APPROVED** [review]
lgtm and CI is happy

**Pull request merged**